File: src/main/java/com/derongan/minecraft/guiy/gui/Pair.java
package com.derongan.minecraft.guiy.gui;

import com.google.auto.value.AutoValue;

/**
 * Simple immutable Pair class that holds two elements.
 *
 * @param <E> The type of element to hold.
 */
@AutoValue
public abstract class Pair<E> {
    public abstract E getFirst();

    public abstract E getSecond();

    public static <E> Pair<E> create(E newFirst, E newSecond) {
        return new AutoValue_Pair<>(newFirst, newSecond);
    }
}

File: src/main/java/com/derongan/minecraft/guiy/gui/Pair.kt
package com.derongan.minecraft.guiy.gui

/**
 * Simple immutable Pair class that holds two elements.
 *
 * @param <E> The type of element to hold.*/
class Pair<E>(val first: E, val second: E) {
    companion object {
        @JvmStatic
        fun <E> create(newFirst: E, newSecond: E) = Pair(newFirst, newSecond)
    }
}